xds/internal/clients/xdsclient/ads_stream.go

```diff
@@ -68,9 +68,10 @@ type dataAndErrTuple struct {
 // occur on the ADS stream. Methods on this interface may be invoked
 // concurrently and implementations need to handle them in a thread-safe manner.
 type adsStreamEventHandler interface {
-	onStreamError(error)                           // Called when the ADS stream breaks.
-	onWatchExpiry(ResourceType, string)            // Called when the watch timer expires for a resource.
-	onResponse(response, func()) ([]string, error) // Called when a response is received on the ADS stream.
+	onStreamError(error)                                       // Called when the ADS stream breaks.
+	onWatchExpiry(ResourceType, string)                        // Called when the watch timer expires for a resource.
+	onResponse(response, func()) ([]string, error)             // Called when a response is received on the ADS stream.
+	onRequiredToRemoveUnsubscribedCacheEntries(typeURL string) // Called when it is needed to remove unsubscribed cache entries.
```

Comment:
How about a shorter, more generic name? Maybe `onRequestSent` or `onRequest` or `onSend`? I like `onSend`, but I think `onRequest` would be better since it goes well with the existing `onResponse`. Thoughts?